documentation/asciidoc/computers/remote-access/network-boot-ipv6.adoc

```diff
@@ -0,0 +1,340 @@
+----
+
+This sends a router solicitation to your router asking for your network details such as the network prefix, router ethernet address and whether to use DHCP for addressing. If there's no response to this command it's likely your network and ISP only supports IPV4. If IPV6 is supported it's most likely that it will be configured to use stateless configuration where clients generate their own addresses.
+
+----
+Soliciting ff02::2 (ff02::2) on eth0...
+Hop limit                 :           64 (      0x40)
+Stateful address conf.    :           No
+Stateful other conf.      :          Yes
+Mobile home agent         :           No
+Router preference         :       medium
+Neighbor discovery proxy  :           No
+Router lifetime           :          180 (0x000000b4) seconds
+Reachable time            :  unspecified (0x00000000)
+Retransmit time           :  unspecified (0x00000000)
+----
+
+You might be able to configure your router for stateful configuration, in which means it will use DHCP to obtain an IP address.
+
+----
+Hop limit                 :           64 (      0x40)
+Stateful address conf.    :          Yes
+Stateful other conf.      :          Yes
+Mobile home agent         :           No
+Router preference         :       medium
+Neighbor discovery proxy  :           No
+Router lifetime           :          180 (0x000000b4) seconds
+Reachable time            :  unspecified (0x00000000)
+Retransmit time           :  unspecified (0x00000000)
+----
+
+=== Debugging
+
+==== Logs and Traces
+
+If the boot uart is enabled you should see something like this from the serial port. The lines starting RX6 indicate that IPV6 is in use.
+
+Here `dc:a6:32:6f:73:f4` is the mac address of the TFTP server and it has an IPV6 address of `fd49:869:6f93::1`. The device itself has a mac address `e4:5f:01:20:24:0b` and an IPV6 address of `fd49:869:6f93::1000`
+
+----
+Boot mode: NETWORK (02) order f
+GENET: RESET_PHY
+PHY ID 600d 84a2
+NET_BOOT: e4:5f:01:20:24:0b wait for link TFTP6: (null)
+LINK STATUS: speed: 100 full duplex
+Link ready
+GENET START: 64 16 32
+GENET: UMAC_START 0xe45f0120 0x240b0000
+RX6: 12 IP: 1 MAC: 1 ICMP: 1/1 UDP: 0/0 ICMP_CSUM_ERR: 0 UDP_CSUM_ERR: 0
+NET fd49:869:6f93::1000 tftp fd49:869:6f93::1
+RX6: 17 IP: 4 MAC: 4 ICMP: 2/2 UDP: 2/2 ICMP_CSUM_ERR: 0 UDP_CSUM_ERR: 0
+TFTP_GET: dc:a6:32:6f:73:f4 fd49:869:6f93::1 ab5a4158/start4.elf
+
+RX6: 17 IP: 4 MAC: 4 ICMP: 2/2 UDP: 2/2 ICMP_CSUM_ERR: 0 UDP_CSUM_ERR: 0
+RX6: 18 IP: 5 MAC: 5 ICMP: 2/2 UDP: 3/3 ICMP_CSUM_ERR: 0 UDP_CSUM_ERR: 0
+TFTP_GET: dc:a6:32:6f:73:f4 fd49:869:6f93::1 ab5a4158/config.txt
+----
+
+Finally the bootloader hands over to firmware which should load the kernel.
+
+==== Stateful configuration
+You can examine network activity with tcpdump.
+
+[,bash]
+----
+sudo tcpdump -i eth0 -e ip6 -XX -l -v -vv
+----
+
+Below is an extract of a tcp dump where the router is configured to use stateful (DHCP) network configuration.
+
+Device sends a router solicitation.
```

Comment:
I would put colons at the end of these, rather than a full stop, to make it a bit more obvious they refer to the extracts which follow.